lib/transaction.js

```diff
@@ -211,30 +223,19 @@ class Transaction extends EventEmitter {
       } catch (e) {
         reject(e);
       }
-    })
-      .then((connection) => {
+    }).then(async (connection) => {
+      try {
         connection.__knexTxId = this.txid;
-
-        return (
-          this._previousSibling
```

Comment:
Short story: `Transaction#acquireConnection(..)` is no longer responsible for waiting on `this._previousSibling`.  This responsibility has been moved to `Transaction#_kickThingsOff(..)` so that its enforcement can be guaranteed across `Transaction` subclasses.